pkg/exchange/okex/exchange.go

```diff
@@ -411,6 +416,124 @@ func (e *Exchange) QueryOrderTrades(ctx context.Context, q types.OrderQuery) ([]
+		return nil, fmt.Errorf("query closed order rate limiter wait error: %w", err)
+	}
+
+	var lastOrder string = strconv.FormatUint(lastOrderID, 10)
+
+	res, err := e.client.NewGetOrderHistoryRequest().
+		InstrumentID(toLocalSymbol(symbol)).
+		StartTime(since).
+		EndTime(until).
+		Limit(defaultQueryLimit).
+		After(lastOrder).
+		Do(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to call get order histories error: %w", err)
+	}
+
+	var orders []types.Order
+
+	for _, order := range res {
+		o, err2 := toGlobalOrder(&order)
+		if err2 != nil {
+			err = multierr.Append(err, err2)
+			continue
+		}
+
+		if o.Status.Closed() {
+			orders = append(orders, *o)
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return types.SortOrdersAscending(orders), nil
+}
+
+/*
+OKEX do not provide api to query by tradeID, So use /api/v5/trade/orders-history-archive as its official site do.
+Query trades can query trades in last 3 months, there are no time interval limitations, as long as end_time >= start_time.
+Please Use LastTradeID as cursor, only return trades earlier than that trade, that trade is not included.
+If you want to query trades by time range, please just pass start_time and end_time.
+If you want to query by cursor, please pass LatestTradeID.
+Because it get correct response even we pass all parameters with right time interval and invalid trade id, like 0.
+*/
+func (e *Exchange) QueryTrades(ctx context.Context, symbol string, options *types.TradeQueryOptions) ([]types.Trade, error) {
+	if symbol == "" {
+		return nil, ErrSymbolRequired
+	}
+
+	req := e.client.NewGetOrderHistoryRequest().InstrumentID(toLocalSymbol(symbol))
+
+	limit := uint64(options.Limit)
+	if limit > defaultQueryLimit || limit <= 0 {
+		log.Debugf("limit is exceeded default limit %d or zero, got: %d, Do not pass limit", defaultQueryLimit, options.Limit)
+	} else {
+		req.Limit(limit)
+	}
+
+	if err := tradeRateLimiter.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("query trades rate limiter wait error: %w", err)
+	}
+
+	var lastTradeID string = strconv.FormatUint(options.LastTradeID, 10)
+
+	var response []okexapi.OrderDetails
+	// query by time interval
+	if options.StartTime != nil || options.EndTime != nil {
+		if options.StartTime != nil {
+			req.StartTime(*options.StartTime)
+		}
+		if options.EndTime != nil {
+			req.EndTime(*options.EndTime)
+		}
+
+		res, err := req.Do(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to call get order histories error: %w", err)
+		}
+		response = res
+	} else if options.StartTime == nil && options.EndTime == nil {
+		res, err := req.Do(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to call get order histories error: %w", err)
+		}
+		response = res
+	} else { // query by trade id
+		res, err := req.Do(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to call get order histories error: %w", err)
+		}
+		for _, trade := range res {
+			if trade.LastTradeID == lastTradeID {
```

Comment:
The last trade id is a cursor, so you should use `before` and pass lastTradeID. Am i right? @c9s

Reply:
e.q. `req.Before(lastTradeId).Do(ctx)`

Reply:
sounds like so